### ISSUE TITLE
BLtouch compensates poorly and room not in the center of the plate to print

### ISSUE THREAD

**Issue by Tam-Koowan** [MarlinFirmware/Marlin#7452]
I have a geeetech Prusa I3 pro B with a BLtouch, with the original version of the Marlin (1.0.2) nothing worked properly
So I'm going to go under "Marlin-bugfix-1.1.x_e16743d-PI3_Pro_B_3DTouch", since when I print my piece is not in the center of the board, but I also my nozzle is crushed on the left and too hasty on the Right of the plate;
I do not know how to solve this problem despite looking for forums.

Thank you for your help.

PS / if necessary I can provide you with photos or videos

[Marlin-bugfix-1.1.x_e16743d-PI3_Pro_B_3DTouch.zip](https://github.com/MarlinFirmware/Marlin/files/1205679/Marlin-bugfix-1.1.x_e16743d-PI3_Pro_B_3DTouch.zip)

**Comment by Roxy-3D** [comment]
You can control where you piece is printed in your Slicer also.    If the reason you are trying to print something small in the center of the bed is because you are trying to bring up UBL, being offset a little is OK.

**Comment by RalphSch** [comment]
Marlin 1.1.x now uses two separated coordinate systems, one for your machine (the real mechanical distance from the endstop to the end of the axis) and one for your workspace.
You can easily switch to the old behavior with NO_WORKSPACE_OFFSET in Configuration_adv.h. If you like to stay with the new system, then you have to travel to your workspace (bed) 0/0/0 (front, left corner, nozzle just touching the print surface) and hit "Set homeoffset" in the lcd menu.

@thinkyhead This is the worst issue when upgrading from 1.0.x to 1.1.x (beside of the by default disabled safe-z-homing which always leads to crashes when the sensor ist mounted in front or left of the nozzle). Allmost all people are crushing their nozzles into the bed and having big problems to get this fixed. Also the "Err: Too far" error when setting the homeoffset is quite confusing ... you have to search through the code to find this very important statement:

M428: Set home_offset based on the distance between the
       current_position and the nearest "reference point."
       If an axis is past center its endstop position
       is the reference-point. Otherwise it uses 0. This allows
       the Z offset to be set near the bed when using a max endstop.
       M428 can't be used more than 2cm away from 0 or an endstop.
       Use M206 to set these values directly.

**Comment by fiveangle** [comment]
@RalphSch - The coordinate systems are aligned unless the user changes them so why do you think home offsets is relevant here ?  

It seems haphazard to throw in using M206 to try and resolve what appears to be more likely a configuration/usage problem.

@Tam-Koowan - what is output of this command:

 `M503 S1`

**Comment by RalphSch** [comment]
@fiveangle Because I help fixing exact this problem more than once every day in our Anet/Prusa FB group (very active, about 5k people) since we migrated the Skynet3D changes for Anet back to Marlin. Both coordinate systems are only aligned by default if your endstops are sitting at the 0 position and that's not given with the Anet printers and many others. This happens when you are upgrading, like Tom (his bed is 12mm away from the x min endstop), and you just grab your working settings from your old Configuration.h and put them in the new Configuration.h. Also you just have to hit the "set home offset" entry in the lcd menu by accident on your home-point (which normally isn't Z0) and your z-home-offset is wrong (this entry should be placed in the configuration section of the menu not in the prepare section next to homing and movement). The "hidden" 20mm max for the distance to 0 or the endstop is also an bad idea, e.g. Anet A8 has often more than this distance and because of that you can't just set your correct home offsets because it aborts with"Err: Too far!". 
If you just activate NO_WORKSPACE_OFFSET then your old and good working settings will work with 1.1.x as before .... that's what normal users prefer when upgrading so this would be the better default (and because the actual situation leads to crashes, bended rods and broken printer, like the disabled z_safe_homing default does when your sensor is located in the front or left of the nozzle).

I'm curios to see the M503 output ;)

**Comment by Roxy-3D** [comment]
> If you just activate NO_WORKSPACE_OFFSET then your old and good working settings will work with 1.1.x as before .... that's what normal users prefer when upgrading so this would be the better default 

I think I agree this is a better default.   It would be helpful to hear from people that don't agree with this thinking why they see it differently.

**Comment by Tam-Koowan** [comment]
12:59:36.557 : N30 M105*20
12:59:36.573 : echo:  G21 ; Units in mm
12:59:36.573 : echo:  M149 C ; Units in Celsius
12:59:36.573 : echo:Filament settings: Disabled
12:59:36.573 : echo:  M200 D3.00
12:59:36.573 : echo:  M200 D0
12:59:36.573 : echo:Steps per unit:
12:59:36.573 : echo:  M92 X80.24 Y80.41 Z2560.00 E97.00
12:59:36.573 : echo:Maximum feedrates (units/s):
12:59:36.573 : echo:  M203 X400.00 Y400.00 Z2.00 E45.00
12:59:36.573 : echo:Maximum Acceleration (units/s2):
12:59:36.573 : echo:  M201 X5000 Y5000 Z50 E5000
12:59:36.573 : echo:Acceleration (units/s2): P<print_accel> R<retract_accel> T<travel_accel>
12:59:36.573 : echo:  M204 P2000.00 R1000.00 T1000.00
12:59:36.573 : echo:Advanced: S<min_feedrate> T<min_travel_feedrate> B<min_segment_time_ms> X<max_xy_jerk> Z<max_z_jerk> E<max_e_jerk>
12:59:36.573 : echo:  M205 S0.00 T0.00 B20000 X10.00 Y10.00 Z0.40 E5.00
12:59:36.573 : echo:Home offset:
12:59:36.573 : echo:  M206 X0.00 Y0.00 Z0.00
12:59:36.573 : echo:Auto Bed Leveling:
12:59:36.573 : echo:  M420 S0 Z0.00
12:59:36.573 : echo:Material heatup parameters:
12:59:36.573 : echo:  M145 S0 H200 B60 F170
12:59:36.573 : M145 S1 H240 B110 F170
12:59:36.573 : echo:PID settings:
12:59:36.573 : echo:  M301 P12.33 I0.51 D74.50
12:59:36.573 : echo:  M304 P234.88 I42.79 D322.28
12:59:36.573 : echo:Z-Probe Offset (mm):
12:59:36.573 : echo:  M851 Z-3.38
12:59:39.599 : N31 M105*21

**Comment by fiveangle** [comment]
Now that we've put that to rest…

I think we need to see a pic of the whole bot from front looking down the bed edge with the carriage close to the bed, and the video of the problem you mentioned.

**Comment by Tam-Koowan** [comment]
![20170719_183458](https://user-images.githubusercontent.com/25410214/29239410-5c1e016e-7f4e-11e7-9946-21dc7418c4c8.jpg)
This is my problem !  :(

**Comment by fiveangle** [comment]
@Tam-Koowan - a few things: 

* You need to enable bed levelling every time you want to use it with `M420 S1`: is that being done ?
* Assuming you're doing that, what is output of `M420 V` to see the stored leveling data ?
* Assuming all that looks good, enable `Z_MIN_PROBE_REPEATABILITY_TEST`, compile, run `M48`, and show us the output

Do you have that picture looking straight-on from the front viewing across the bed edge with the carriage close to it so you can see the relation of the cariage with the frame/bed ?

**Comment by Tam-Koowan** [comment]
Cette photos à été prise depuis l'avant de l'imprimante. 
Je tester ça en rentant car je suis en week-end chez des amis  


This picture was taken from the front of the printer.
I test this by coming back because I'm at weekends with friends

_Sent from my Samsung SM-G800F using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Roxy-3D** [comment]
It looks like you bed has a very significant tilt to it.   It would be wise to correct that as much as possible.

**Comment by Tam-Koowan** [comment]
@Roxy-3D :Yes, this is the problem I have, my BLtouch does not compensate as it shoul,
my bed level is good.

@fiveangle: 23:34:08.124 : echo:Bed Leveling On
23:34:10.300 : N19 M105*31
23:34:13.355 : N20 M105*21
23:34:16.412 : N21 M105*20
23:34:19.466 : N22 M105*23
23:34:22.528 : N23 M105*22
23:34:25.585 : N24 M105*17
23:34:28.639 : N25 M105*16
23:34:31.702 : N26 M105*19
23:34:34.722 : N27 M420 V*102
23:34:34.747 : Bilinear Leveling Grid:
23:34:34.747 : 0      1      2      3
23:34:34.747 : 0 +0.109 +0.234 +0.428 +0.577
23:34:34.747 : 1 +0.063 +0.262 +0.470 +0.644
23:34:34.747 : 2 -0.025 +0.216 +0.452 +0.699
23:34:34.747 : 3 -0.128 +0.226 +0.513 +0.780
23:34:34.747 : echo:Bed Leveling On

**Comment by Roxy-3D** [comment]
OK...   But this picture doesn't make it look like the physical bed leveling is good.  Was bed leveling turned on when this picture was taken?   https://github.com/MarlinFirmware/Marlin/issues/7452#issuecomment-321967990

**Comment by Tam-Koowan** [comment]
@Roxy-3D : It is not systematically activated in the "Marlin-bugfix-1.1.x_e16743d-PI3_Pro_B_3DTouch"?
If not I do not know if it is!
It seems to me that my physical level of the bed is good, I followed the Geeetech method to set it.

**Comment by Roxy-3D** [comment]
Look at your grid numbers.  You have the right side of the mesh 1mm higher than the left side.

It is possible you got that bad print you posted up above because you did not turn on the bed leveling system.   So no correction was happening for the 1mm tilt you have.

**Comment by Tam-Koowan** [comment]
So I have to go up my bed 1 mm on the left side to test?

**Comment by Roxy-3D** [comment]
- Activate the bed leveling...
- lower the nozzle to 1.5mm or 2mm above the bed
- Make some big moves from left to right or front to back.  You should see the Z-Axis compensate if the system is alive and working.

1mm is a lot of tilt.  You really should do what you can to minimize that tilt.   But with that said, the bed leveling system can compensate for that.   It would do a better job if the tilt was less.   But the system can handle it.

**Comment by Tam-Koowan** [comment]
I have to put the M420 S1 code in my boot code I guess, but I have to put it before the G29?

**Comment by Roxy-3D** [comment]
After the G29.

**Comment by Tam-Koowan** [comment]
Ok thank you, I test right away

**Comment by Roxy-3D** [comment]
@fiveangle    I left a short message for you over at 3DPrintboard.com

**Comment by Tam-Koowan** [comment]
Well I returned from vacation, and I wanted to re-download the bugfix marlin ... in my printer and the I have an error code, except that I did not touch the setting in the marlin.
Here's the code:  'INVERT_X_DIR' was not declared in this scope


You would not have a petrol on you and a match? :(

**Comment by Roxy-3D** [comment]
It sounds like a ; got deleted somewhere in the configuration.h file...

**Comment by Tam-Koowan** [comment]
Ok except that I did not touch the file, it is the same as I provided you in the Zip above ...
So I do not understand

**Comment by Roxy-3D** [comment]
You deleted an important line of the configuration.h file:

![image](https://user-images.githubusercontent.com/8404322/30008053-5430510c-90e0-11e7-891b-6ea1d6f6baa0.png)

**Comment by Tam-Koowan** [comment]
Well, I managed to put the marlin back in my printer.
Except that I have nothing to this line before downloading it.
So what do you think is the best method of leveling the bed and how do I activate it?

**Comment by Roxy-3D** [comment]
Because Y is inverted...  It is more likely X is inverted than not.   Just pick a true or false value for
```cpp
#define INVERT_X_DIR true
```
You will know as soon as you try to move the X axis if it is moving in the right direction.

**Comment by Tam-Koowan** [comment]
What method of bed leveling should I use?
for the best result.
My problem of centering of plateau will be solved because I find the solution on a French Forum ;)

**Comment by Roxy-3D** [comment]
I wrote the UBL (Unified Bed Leveling) System.   I'm not sure I would be considered an objective source.

If you want my assistance with the bed leveling issues...  You need to be running UBL.

**Comment by Tam-Koowan** [comment]
Very well, I would change the settings of my marlin tomorrow for tried.
There just turn on or there are parameters to be set?
thank you very much

**Comment by Roxy-3D** [comment]
Make sure you can print a small object at the center of the bed before you turn on any bed leveling system.   Once you can do that,  turn these options on.  Probably the existing values are fine for you.
```cpp
#define AUTO_BED_LEVELING_UBL
 //===========================================================================
  //========================= Unified Bed Leveling ============================
  //===========================================================================

  #define UBL_MESH_INSET 1          // Mesh inset margin on print area
  #define UBL_MESH_NUM_X_POINTS 10  // Don't use more than 15 points per axis, implementation limited.
  #define UBL_MESH_NUM_Y_POINTS 10
  #define UBL_PROBE_PT_1_X 39       // These set the probe locations for when UBL does a 3-Point leveling
  #define UBL_PROBE_PT_1_Y 180      // of the mesh.
  #define UBL_PROBE_PT_2_X 39
  #define UBL_PROBE_PT_2_Y 20
  #define UBL_PROBE_PT_3_X 180
  #define UBL_PROBE_PT_3_Y 20
  #define UBL_G26_MESH_EDITING    // Enable G26 mesh editing
```

**Comment by Tam-Koowan** [comment]
After testing the compilation it puts this error:
The given UBL_PROBE_PT_1_Y can not be reached by the Z probe.

I have therefore modified:


  #define UBL_MESH_INSET 1 // Mesh inset margin on print area
   #define GRID_MAX_POINTS_X 10 // Do not use more than 15 points per axis, implementation limited.
   #define GRID_MAX_POINTS_Y GRID_MAX_POINTS_X
   #define UBL_PROBE_PT_1_X 39 // UBL does a 3-Point leveling
   #define UBL_PROBE_PT_1_Y 130 // (Old to 180) of the mesh.
   #define UBL_PROBE_PT_2_X 39
   #define UBL_PROBE_PT_2_Y 20
   #define UBL_PROBE_PT_3_X 180
   #define UBL_PROBE_PT_3_Y 20
   #define UBL_G26_MESH_EDITING // Enable G26 mesh editing

I think I have to change it because of the position of my BLtouch?

**Comment by Roxy-3D** [comment]
Yes.   And those values will probably never get used by you.  They are just there in case you want to do a 3-point tilting of the mesh.

**Comment by Tam-Koowan** [comment]
But why did I get this error when I activated this function?

**Comment by Roxy-3D** [comment]
Because you have a large offset from the nozzle to where the probe is located.   The nozzle can make it to that location but the probe can not go there.   So that Sanity Check code detected the problem and informed you about it so that it could get fixed.

**Comment by Tam-Koowan** [comment]
I no longer have any error when compiling.
But when I run an impression, the sensor does not activate, as if I had the original confirmation?
You have an idea ?
Here is a video of what printing gives
https://drive.google.com/open?id=0B3nheGhaAmwpRUdnekdIamJCaXc

**Comment by Roxy-3D** [comment]
Have you done a G26 Mesh Validation Pattern?   Can you print one and post the picture of it?   That will help us know what is out of adjustment.    It is hard to tell from the video but it looks like the nozzle is too far from the bed.   Also...  Please post a G29 W    It looks like UBL is not activated.   (You do that with G29 A)     Be sure to activate UBL with a G29 A prior to doing the G26 Mesh Validation Pattern.

**Comment by Tam-Koowan** [comment]
After putting back my Offset Z I made several impressions with success despite the lag between the two sides. But after 4 good impression I got an error message.
I think my BLtouch is dead?
![image](https://user-images.githubusercontent.com/25410214/30253148-8bef6412-967f-11e7-8905-6db14dbf13e2.png)

So I put back the Z's mini level sensor and I re-done the setting with the sheet, but I can not print correctly
